lib/Sema/TypeCheckDecl.cpp

```diff
@@ -482,6 +482,53 @@ static void checkInheritanceClause(
   }
 }
 
+// Check for static properties that produce empty option sets
+static void checkForEmptyOptionSet(const VarDecl *VD, TypeChecker &tc) {
+  if (!VD->isStatic() || !VD->isLet())
+    return;
+  
+  auto DC = VD->getDeclContext();
+  
+  if (!VD->getType()->isEqual(DC->getSelfTypeInContext()))
+    return;
+  
+  auto *optionSetProto = tc.Context.getProtocol(KnownProtocolKind::OptionSet);
+  auto protocolConformance = !tc.containsProtocol(
+                                                  DC->getSelfTypeInContext(),
+                                                  optionSetProto,
+                                                  DC,
+                                                  /*Flags*/None);
+  
+  if (!protocolConformance)
+    return;
+  
+  auto PBD = VD->getParentPatternBinding();
+  if (!PBD)
+    return;
+  
+  for (auto entry : PBD->getPatternList()) {
```

Comment:
You should be able to get the entry with `PBD->getPatternEntryForVarDecl(VD)` instead of searching through the list yourself.

Reply:
Oops, I forgot about this API. This one's my bad. ><